src/Workspaces/Core/Portable/CodeGeneration/AbstractFlagsEnumGenerator.cs

```diff
@@ -73,12 +71,13 @@ private SyntaxNode CreateFlagsEnumConstantValue(INamedTypeSymbol enumType, objec
             return CreateFlagsEnumConstantValue(enumType, constantValue, allFieldsAndValues, usedFieldsAndValues);
         }
 
-        private SyntaxNode CreateFlagsEnumConstantValue(
+        private SyntaxNode? CreateFlagsEnumConstantValue(
             INamedTypeSymbol enumType,
             object constantValue,
             List<(IFieldSymbol field, ulong value)> allFieldsAndValues,
             List<(IFieldSymbol field, ulong value)> usedFieldsAndValues)
         {
+            Contract.ThrowIfNull(enumType.EnumUnderlyingType);
```

Comment:
Do we need to be handling this somehow? (I agree this PR isn't making it any worse, but this seems surprising...)

Reply:
A higher layer has ensured that this is an enum type.  I'm of the belief that this will always have a .EnumUnderlyingType for those:


```
        /// <summary>
        /// For enum types, gets the underlying type. Returns null on all other
        /// kinds of types.
        /// </summary>
        INamedTypeSymbol? EnumUnderlyingType { get; }
```